2_process/src/munge_inst_timeseries.R

```diff
@@ -55,3 +55,26 @@ aggregate_data_to_hourly <- function(inst_data,output_tz){
 
 }
 
+aggregate_data_to_daily <- function(inst_data, daily_data){
+  #' 
+  #' @description Function to aggregate instantaneous NWIS data collected at sub-hourly (e.g. 15/5/30 min) intervals to hourly min/mean/maxs
+  #'
+  #' @param inst_data a data frame containing the downloaded time series for NWIS instantaneous site. 
+  #' inst_data must include the following columns: c("Value_Inst","dateTime","site_no")
+  #' @param daily_data a data frame the downloaded daily time series of DO data. This is used so that if a site is already in 
+  #'  the daily sites, we won't do the aggregating here
+  #'
+  #' @value A data frame containing daily min, mean, and max values the parameter of interest
+  
+  
+  only_inst_data = setdiff(inst_data$site_no, daily_data$site_no)
```

Comment:
Here we're only calculating daily summaries for instantaneous sites that don't report daily summaries (n=15 SC sites right now). That makes sense, since I think we'll want to default to the reported daily values whenever they are available. However, each site would have to report each of the stat_cd's we're interested in - mean/max for inland-salinity. It looks like most of the daily sites do report mean/min/max but at least one SC site (01473169) only reports mean/min. So we would either forgo the max data for that site, or calculate the daily summaries and handle the uv/dv selection in your `p2_daily_combined` target. Do you have any thoughts on that?

Reply:
I think that the padding can happen downstream of this step. At least in past work it has happened downstream, closer to where the actual dynamic modeling happens.

Reply:
We ultimately want daily mean/max/(min) for all continuous sites. We could take two approaches: 1) assume that the local data contributors have already QA/QC'ed the daily records. To combine the daily-aggregated uv values with the daily data in `p1_daily_data` we could then just take a diff approach: use daily summaries for all dv sites and then just add our inst daily summaries for the ~15 inst only sites. 2) Alternatively, we could try to fill in any gaps in the daily record with our daily summaries but that means assuming the reporting offices didn't leave that gap there for a good reason. 

@jds485 do you have an opinion on that? I think Jeff and I are leaning toward the first option, but we thought you may want to weigh in as well.

Reply:
Just so to make sure I'm understanding the suggestion, daily records = daily summaries = daily mean, min, and max? Did you say earlier this week that you compared dv daily records to equivalent obtained from the uv data and found them to be very similar? If so, approach 1 sounds good to me. 

As an extension of that logic, if we trust the QAQC is complete for dv, then should we drop uv data on the days there are gaps in the daily record? For example, if a day has a complete uv record, it would get a dv entry if the uv data were good. So a missing dv entry would indicate that something was wrong with the uv data and we shouldn't use it. What do you think?

Reply:
With approach 1, we wouldn't even be calculating daily summaries for `dv` sites, just for `uv` sites. So there's no chance, if the code is doing what I think it is, that we would be "filling in" missing `dv` data with `uv` summaries.

Reply:
Also, did you see my [comment](https://github.com/USGS-R/drb-inland-salinity-ml/pull/26#issuecomment-995273276) below about using hourly aggregated data instead of uv? It looks like you've retained the NAs, so it may not be a helpful suggestion anymore

Reply:
Yeah, we decided to aggregate the uv data instead of the hourly-aggregated data because 1) the local data contributors use the uv values when reporting the daily summaries, so using the uv data would be more comparable with the dv data, and 2) because we thought using the hourly-aggregated values would potentially misrepresent the daily extremes (min/max) that we're interested in.

Reply:
Okay, that makes sense to use uv peaks, then.

for 2, yes I agree. Using hourly data is a different interpretation (maximum of hourly averages instead of daily maximum), and that's less useful for matching to the WQ criteria. This got me thinking about over what timescale SC and DO change. Pulse road salt events will change SC rapidly. Are there similar pulse events for DO? No need to answer here, just something to be thinking about - are our sampling frequencies matching the timescales of expected WQ changes?

Reply:
> Just so to make sure I'm understanding the suggestion, daily records = daily summaries = daily mean, min, and max? Did you say earlier this week that you compared dv daily records to equivalent obtained from the uv data and found them to be very similar? If so, approach 1 sounds good to me.

Oof, too many descriptors, but yes I'm using those terms equivalently here. I did previously compare the dv daily records to the equivalent values calculated from the uv data (for the 80 sites that have inst and daily data) and they are very similar: 

![compare_uv_and_dv](https://user-images.githubusercontent.com/8785034/146587995-2a7f4eda-3795-4c5e-ad59-d71fc375e489.png)